src/NLog/LayoutRenderers/LevelLayoutRenderer.cs

```diff
@@ -71,15 +91,23 @@ protected override void Append(StringBuilder builder, LogEventInfo logEvent)
                     break;
                 case LevelFormat.FullName:
                     if (level == LogLevel.Info)
-                        builder.Append("Information");
+                        builder.Append(Uppercase ? "INFORMATION" : "Information");
                     else if (level == LogLevel.Warn)
-                        builder.Append("Warning");
+                        builder.Append(Uppercase ? "WARNING" : "Warning");
                     else
-                        builder.Append(level.ToString());
+                        builder.Append(Uppercase ? GetUpperCaseString(level) : level.ToString());
                     break;
             }
         }
 
+        private string GetUpperCaseString(LogLevel level)
+        {
+            if (_upperCaseMapper.TryGetValue(level.Ordinal, out var uppercaseString))
+                return uppercaseString;
+            else
+                return level.ToString().ToUpperInvariant();
```

Comment:
When could this happen?

Reply:
I like fallbacks instead of exceptions.

Reply:
me too, but this is impossible to reach? Dead code is also not nice (untested, could break in the future)

Reply:
If I had implemented some crazy fallback logic, then it would make sense to consider not having it. But this seem very harmless to me.

Reply:
Well its confusing and it wont have a goal? You could argue this will prevent bugs in the future, but I think it will lead to bugs in the future as it's impossible to test.